networking/podenv.go

```diff
@@ -65,31 +61,21 @@ type activeNet struct {
 
 // Loads nets specified by user and default one from stage1
 func (e *podEnv) loadNets() ([]activeNet, error) {
-	nets, err := loadUserNets(e.localConfig, e.netsLoadList)
-	if err != nil {
-		return nil, err
-	}
-
 	if e.netsLoadList.None() {
-		return nets, nil
+		stderr.Printf("networking namespace with loopback only")
+		return nil, nil
 	}
 
-	if !netExists(nets, "default") && !netExists(nets, "default-restricted") {
-		var defaultNet string
-		if e.netsLoadList.Specific("default") || e.netsLoadList.All() {
-			defaultNet = DefaultNetPath
-		} else {
-			defaultNet = DefaultRestrictedNetPath
-		}
-		defPath := path.Join(common.Stage1RootfsPath(e.podRoot), defaultNet)
-		n, err := loadNet(defPath)
-		if err != nil {
-			return nil, err
-		}
-		nets = append(nets, *n)
+	nets, err := e.newNetLoader().loadNets(e.netsLoadList)
+	if err != nil {
+		return nil, err
+	}
+	netSlice := make([]activeNet, 0, len(nets))
+	for _, net := range nets {
```

Comment:
ah, you are right!

The logic here can become very complex. We either use the order defined by `.conf` file names on disk, with some other logic to merge contents from `/etc/rkt/net.d` and what's built into stage1, or we use the order from the command line, but it is unclear what to do for `--net=all`.

Thoughts?